frontend/src/pages/editCategory.js

```diff
@@ -1,27 +1,60 @@
-import React, { useContext, useEffect, useState } from "react";
+import React, { useEffect, useState } from "react";
 import Container from "react-bootstrap/Container";
-import Button from "react-bootstrap/Button";
-import Form from "react-bootstrap/Form";
-import Row from "react-bootstrap/Row";
-import Col from "react-bootstrap/Col";
-import InputField from "../components/input";
+import CategoryForm from "../components/categoryForm";
+import { useNavigate, useParams } from "react-router-dom";
+import { editCategory, getCategory } from "../services/adminService";
+import Row from "react-bootstrap/esm/Row";
+import Col from "react-bootstrap/esm/Col";
+
+const EditCategory = () => {
+  const { id } = useParams();
+  const navigate = useNavigate();
+
+  const [state, setState] = useState(null);
+  const [error, setError] = useState();
+  const [success, setSuccess] = useState(false);
+
+  const handleSave = async () => {
+    editCategory(state)
+      .then(() => {
+        setSuccess(true);
+      })
+      .catch(({ response }) => {
+        setError(response.data);
+      })
+      .finally(() => {});
+  };
```

Comment:
You dont need success state coz you only need it once, so you can redirect directly after saving
```suggestion

  const handleSave = async () => {
    editCategory(state)
      .then(() => {
        navigate("/admin/categories")
      })
      .catch(({ response }) => {
        setError(response.data);
      })
  };
```